src/components/password/password.component.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useState } from "react";
+import { TextboxProps } from "../textbox";
+import { StyledPassword, HiddenAriaLive } from "./password.style";
+import guid from "../../__internal__/utils/helpers/guid";
+import tagComponent, {
+  TagProps,
+} from "../../__internal__/utils/helpers/tags/tags";
+import useLocale from "../../hooks/__internal__/useLocale";
+import ButtonMinor from "../button-minor/button-minor.component";
+
+export type TypeOptions = "text" | "password";
+
+export interface PasswordProps extends TextboxProps, TagProps {
+  /** @private @ignore */
+  type?: TypeOptions;
+  /** When `true` input `type` is `password` and text is obscured. */
+  forceObscurity?: boolean;
+}
+
+export const Password = ({
+  id,
+  disabled,
+  forceObscurity = false,
+  ...rest
+}: PasswordProps) => {
+  const internalInputId = useRef(id || guid());
+  const l = useLocale();
+
+  const [passwordShown, setPasswordShown] = useState(false);
+  const visibleInput = !forceObscurity && passwordShown;
+  return (
+    <>
+      <StyledPassword
+        {...tagComponent("password", rest)}
+        data-element="styled-password-container"
+        id={internalInputId.current}
+        autoComplete="off"
+        type={visibleInput ? "text" : "password"}
+        disabled={disabled}
+        {...rest}
+      >
+        <ButtonMinor
```

Comment:
Given `forceObscurity` overrides the control of this button, we will either need to disable it or not render it, since this could be confusing for users if they attempt to press the button and expect the visibility to change.

Reply:
Went for `disabled` to ensure we stay within designs 👍